front/src/applications/editor/Editor.tsx

```diff
@@ -185,6 +195,81 @@ const Editor = () => {
     }
   }, []);
 
+  /**
+   * When the component mounts
+   * => get the searchParams
+   * => if theere is a selection param, select the entities and focus on them
+   */
+  useEffect(() => {
+    if (urlInfra) {
+      const params = searchParams.get('selection');
+      if (!params && searchParams.size !== 0) {
+        dispatch(
+          setFailure({
+            name: t('translation:Editor.tools.select-items.errors.unable-to-select'),
+            message: t('translation:Editor.tools.select-items.errors.invalid-url'),
+          })
+        );
+        navigate(`/editor/${urlInfra}`);
+      }
+      const paramsList = params?.split('|');
+      if (paramsList && paramsList.length === 1) {
+        const [objType, entityId] = paramsList[0].split('~');
+        const selectObjectAndFocus = async () => {
```

Comment:
I think it would be nice also to factorize code like this, to have only 1 async function (also typo selectObjetcsAndFocus => selectObjectsAndFocus)
```ts
        const selectObjectsAndFocus = async (
            entitiesInfos: { id: string; type: EditoastType }[]
          ) => {
            let entities: EditorEntity[];
            if (!entitiesInfos.length) return;
            if (entitiesInfos.length === 1) {
              const { type: objType, id: entityId } = selectedEntities[0];
              const entity = await getEntity(+urlInfra, entityId, objType as ObjectType, dispatch);
              entities = [entity];
            } else {
              const entitiesRecord = await getMixedEntities(+urlInfra, entitiesInfos, dispatch);
              entities = Object.values(entitiesRecord);
            }

            selectEntities(entities, { switchTool, dispatch, editorState });

            if (mapRef.current) centerMapOnObject(+urlInfra, entities, dispatch, mapRef.current);
          };

          selectObjetcsAndFocus(selectedEntities);
```

Reply:
Done